lib/modem_info/modem_info.c

```diff
@@ -577,39 +589,53 @@ int modem_info_string_get(enum modem_info info, char *buf,
 		}
 	}
 
-	if ((info == MODEM_INFO_IP_ADDRESS) && (ip_cnt > 0)) {
-		/* for now get only IPv4 address, discard IPv6
-		 * which are separated by a space
-		 */
-		char *ip_v6_str = strstr(&buf[out_buf_len], " ");
+	if (info == MODEM_INFO_IP_ADDRESS) {
+		if ((ip_cnt > 0) && (len > 0)) {
+			/* For now get only IPv4 address, discard IPv6
+			 * which are separated by a space
+			 */
+			char *ip_v6_str = strstr(&buf[out_buf_len], " ");
+
+			if (ip_v6_str) {
+				/* Discard IPv6 info and adjust length */
+				*ip_v6_str = 0;
+				len = strlen(&buf[out_buf_len]);
+			}
 
-		if (ip_v6_str) {
-			/* discard IPv6 info and adjust length */
-			*ip_v6_str = 0;
-			len = strlen(&buf[out_buf_len]);
-		}
-		out_buf_len += len;
-		/* if there are more addresses, add a separator */
-		if (ip_cnt > 1) {
-			err = snprintf(&buf[out_buf_len],
-					buf_size - out_buf_len,
-					IP_ADDR_SEPARATOR);
-			if ((err <= 0) || (err > (buf_size - out_buf_len))) {
-				return -EMSGSIZE;
+			out_buf_len += len;
+
+			/* If there are more addresses, add a separator */
+			if (ip_cnt > 1) {
+				err = snprintf(&buf[out_buf_len],
+						buf_size - out_buf_len,
+						IP_ADDR_SEPARATOR);
+				if ((err <= 0) || (err > (buf_size - out_buf_len))) {
+					return -EMSGSIZE;
+				}
+
+				out_buf_len += IP_ADDR_SEPARATOR_LEN;
+
+				/* Advance (after null) to next IP string */
+				cmd_rsp_idx += ip_str_len + 1;
 			}
-			out_buf_len += IP_ADDR_SEPARATOR_LEN;
-			/* advance (after null) to next IP string */
-			cmd_rsp_idx = ip_str_len + 1;
-		}
 
-		if (--ip_cnt) {
-			goto parse;
-		} else {
-			len = out_buf_len;
+			if (--ip_cnt) {
+				goto parse;
+			} else {
+				len = out_buf_len;
+			}
+		} else if ((ip_cnt > 0) && (len == 0)) {
+			/* The case where an empty IP string was detected */
+			if ((--ip_cnt == 0) && (total_ips > 1)) {
+				/* The last IP was empty, remove stray separator */
+				out_buf_len -= IP_ADDR_SEPARATOR_LEN;
+				buf[out_buf_len] = '\0';
+				len = out_buf_len;
+			}
 		}
 	}
 
-	return len <= 0 ? -ENOTSUP : len;
+	return len <= 0 ? -ENOTSUP : out_buf_len;
```

Comment:
`out_buf_len` is only used for IP parsing.

Reply:
Thanks! 
I also fixed an issue with the newly introduces `accumulated_len` variable so that it's actually reset between each parsing of a new command. It ended writing out of bounds of the buffer, corrupting memory and causing reboot.